bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -0,0 +1,350 @@
+            new Capabilities(4).primary().tiltAnywhere().tilt180()            .text("Vertical Tilt 180°"),
+            new Capabilities(5)          .tiltAnywhere().tilt180()            .text("Tilt Only 180°"),
+            new Capabilities(6).primary()                                     .text("Top Down")                 .primaryStateInverted(),
+            new Capabilities(7).primary()                         .secondary().text("Top Down Bottom Up"),
+            new Capabilities(8).primary()                                     .text("Duolite Lift"),
+            new Capabilities(9).primary().tiltAnywhere()                      .text("Duolite Lift and Tilt 90°"),
+        // @formatter:on
+            new Capabilities() };
+
+    /*
+     * Database of known shade types and corresponding capabilities.
+     */
+    private static final Type[] TYPE_DATABASE = {
+        // @formatter:off
+            new Type( 4).capabilities(0).text("Roman"),
+            new Type( 5).capabilities(0).text("Bottom Up"),
+            new Type( 6).capabilities(0).text("Duette"),
+            new Type( 7).capabilities(6).text("Top Down"),
+            new Type( 8).capabilities(7).text("Duette Top Down Bottom Up"),
+            new Type( 9).capabilities(7).text("Duette DuoLite Top Down Bottom Up"),
+            new Type(23).capabilities(1).text("Silhouette"),
+            new Type(42).capabilities(0).text("M25T Roller Blind"),
+            new Type(43).capabilities(1).text("Facette"),
+            new Type(44).capabilities(0).text("Twist"),
+            new Type(47).capabilities(7).text("Pleated Top Down Bottom Up"),
+            new Type(49).capabilities(0).text("AC Roller"),
+            new Type(51).capabilities(2).text("Venetian"),
+            new Type(54).capabilities(3).text("Vertical Slats Left Stack"),
+            new Type(55).capabilities(3).text("Vertical Slats Right Stack"),
+            new Type(56).capabilities(3).text("Vertical Slats Split Stack"),
+            new Type(62).capabilities(2).text("Venetian"),
+            new Type(69).capabilities(3).text("Curtain Left Stack"),
+            new Type(70).capabilities(3).text("Curtain Right Stack"),
+            new Type(71).capabilities(3).text("Curtain Split Stack"),
+            new Type(79).capabilities(8).text("Duolite Lift"),
+        // @formatter:on
+            new Type() };
+
+    /**
+     * Base class that is extended by Type and Capabilities classes.
+     *
+     * @author Andrew Fiddian-Green - Initial Contribution
+     */
+    private static class Base {
+        protected int intValue = -1;
+        protected String text = "-- not in database --";
+
+        @Override
+        public String toString() {
+            return String.format("%d ( %s )", intValue, text);
+        }
+    }
+
+    /**
+     * Describes a shade type entry in the database; implements 'capabilities' parameter.
+     *
+     * @author AndrewFG - Initial contribution
```

Comment:
Full name missing here, but I think author tag is only needed on file level?

Reply:
I will change it to my full name. I'm not sure, but I think (maybe) it is required on all classes (included embedded ones), rather than just on outer classes?? But anyway it does no harm.